### PR TITLE
added support for ordinal axes

### PR DESCRIPTION
**Pull request by napisani**
added support for ordinal axes

### CODE REVIEW

File: src/utils/index.js [hunk 1]
```diff
@@ -1,14 +1,29 @@
 var d3 = require('d3');
 
 
-exports.calculateScales = (chartWidth, chartHeight, xValues, yValues) => {
+Array.prototype.unique = function() {
+  var o = {}, i, l = this.length, r = [];
+  for(i=0; i<l;i+=1) o[this[i]] = this[i];
+  for(i in o) r.push(o[i]);
+  return r;
+};
```
Comment (mwhite) [inline src/utils/index.js]: It's generally not good to extend native prototypes with non-standard methods in library code.

**Comment by napisani** [comment]
I changed 'unique' to be a helper function. I'd like to re-request a pull.

File: src/utils/index.js [hunk 2]
```diff
@@ -17,7 +31,14 @@ exports.calculateScales = (chartWidth, chartHeight, xValues, yValues) => {
   if (yValues.length > 0 && Object.prototype.toString.call(yValues[0]) === '[object Date]') {
     yScale = d3.time.scale()
       .range([chartHeight, 0]);
-  } else {
+  } else if (yValues.length > 0 && Object.prototype.toString.call(yValues[0]) === '[object String]'){
+    tickDist = (chartHeight) / findUniqueValues(yValues).length;
+    for (tickRangeNum = chartHeight ; tickRangeNum > 0; tickRangeNum -= tickDist) {
+      tickRange.push(tickRangeNum);
+    }
+    console.log(JSON.stringify(tickRange));
```
Comment (yang-wei) [inline src/utils/index.js]: I think you forget to remove this line ?

**Comment by yang-wei** [comment]
> added support for ordinal axes

Your pull request did a lot than this. why did you change back es6 syntax to es5 ?

**Comment by napisani** [comment]
updated.

File: src/utils/index.js [hunk 3]
```diff
@@ -1,14 +1,28 @@
...
 
+var findUniqueValues = function(list) {
+  var o = {}, i, l = list.length, r = [];
+  for(i=0; i<l;i+=1) o[list[i]] = list[i];
+  for(i in o) r.push(o[i]);
+  return r;
+};
 
-exports.calculateScales = (chartWidth, chartHeight, xValues, yValues) => {
+exports.calculateScales = function(chartWidth, chartHeight, xValues, yValues)  {
```
Comment (yang-wei) [inline src/utils/index.js]: indeed you do not have to change these 2 lines.

We prefer using `exports.calculateScales = (chartWidth, chartHeight, xValues, yValues) => {`

File: src/utils/index.js [hunk 4]
```diff
@@ -78,29 +91,17 @@ exports.flattenData = (data, xAccessor, yAccessor) => {
...
-        // in allValues or in xValues or yValues
-        if (isNaN(y)) {
-          return;
-        }
         yValues.push(y);
         yNode = y;
       }
 
-      var xyCoords = `${x}-${yNode}`;
+      var xyCoords = (x + "-" + yNode);
```
Comment (yang-wei) [inline src/utils/index.js]: I am sure 
```
`${x}-${yNode}`
```
is more readable.

File: src/utils/index.js (hunk 3 above)
Comment (napisani) [inline src/utils/index.js]: how about now?

Comment (yang-wei) [inline src/utils/index.js]: you still miss one @napisani

**Comment by napisani** [comment]
updated

File: src/utils/index.js [hunk 5]
```diff
@@ -78,24 +91,12 @@ exports.flattenData = (data, xAccessor, yAccessor) => {
...
           // as in the case of candlestick, default to y value of 0
           yNode = 0;
         });
       } else {
-        // Check for NaN since d3's Voronoi cannot handle NaN values
-        // Go ahead and Proceed to next iteration since we don't want NaN
-        // in allValues or in xValues or yValues
-        if (isNaN(y)) {
-          return;
-        }
```
Comment (yang-wei) [inline src/utils/index.js]: is there any reason you remove all the isNan() check. We need this in graph with nested data.

**Comment by napisani** [comment]
yes, I removed the checks because if you are working with an ordinal axis then your data is inherently no a number. For example, a point on a graph of "Time VS Location" may look like this:
{
    x: new Date(),
    y: "New York"
}

For a graph of this nature, all y values will be NaN, but the changes support having strings as Axis values now.

**Pull request closed without merging**